### PR TITLE
core(doctype): check for limited quirks mode

### LINKED ISSUE #14570: Improve detection of quirks mode
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://www.example.com

### What happened?

This is the response Lighthouse issues for pages with valid legacy doctypes like HTML 4.01 and XHTML 1.0, but all browsers I tested issue 'CS11Compat' (i.e. Standards Mode) for those. So the Lighthouse response is clearly a bug.

### What did you expect?

Accept valid doctypes

### What have you tried?

_No response_

### How were you running Lighthouse?

PageSpeed Insights

### Lighthouse Version

Sorry, no version number given (latest website version)

### Chrome Version

_No response_

### Node Version

_No response_

### OS

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by connorjclark**
fixes #14570

We can't determine limited-quirks-mode without doing a bunch of work ourselves; and the only protocol is via inspector issues which is not snapshot friendly but given this is a very minor edge case, it seems ok to use an optional artifact here. FR doesn't know how to schedule such artifacts, but the runner still passes to the audit when possible, so in the most common cases (not filtering to just doctype audit, or excluding all audits that require InspectorIssues) this should just work. And when it doesn't... whatever?

Without knowing for sure if something is no-quirks-mode, we will always pass the audit.

### CODE REVIEW

File: core/test/audits/dobetterweb/doctype-test.js [hunk 1]
```diff
@@ -72,10 +90,11 @@ describe('DOBETTERWEB: doctype audit', () => {
     expect(auditResult.explanation).toBeDisplayString('Expected systemId to be an empty string');
   });
 
-  it('succeeds when document contains a doctype, and the name value is "html"', () => {
+  // eslint-disable-next-line max-len
+  it('succeeds when document is CSS1Compat, did not detect limit-quirks-mode, and regardless of doctype values seen', () => {
```
Comment (connorjclark) [inline core/test/audits/dobetterweb/doctype-test.js:94]: this test is not very realistic, any better ideas?

File: core/audits/dobetterweb/doctype.js [hunk 2]
```diff
@@ -94,7 +116,7 @@ class Doctype extends Audit {
       };
     } else {
       return {
-        score: 1,
+        score: 0,
```
Comment (adamraine) [inline core/audits/dobetterweb/doctype.js:119]: I think `UIStrings.explanationWrongDoctype` works as a catch-all fallback here, we probably don't need this extra if-else case.

File: core/audits/dobetterweb/doctype.js [hunk 3]
```diff
@@ -41,6 +43,7 @@ class Doctype extends Audit {
       failureTitle: str_(UIStrings.failureTitle),
       description: str_(UIStrings.description),
       requiredArtifacts: ['Doctype'],
+      __internalOptionalArtifacts: ['InspectorIssues'],
```
Comment (adamraine) [inline core/audits/dobetterweb/doctype.js:46]: If we're going to use this outside the experimental config, maybe drop the `__internal`?

File: core/test/audits/dobetterweb/doctype-test.js (hunk 1 above)
Comment (adamraine) [reply core/test/audits/dobetterweb/doctype-test.js:94]: Maybe use the case from the reported issue

File: core/audits/dobetterweb/doctype.js (hunk 3 above)
Comment (connorjclark) [reply core/audits/dobetterweb/doctype.js:46]: Not until FR does a better job re: scheduling. Also that's a bigger change :/

**Review by adamraine — APPROVED** [review]

File: core/test/fixtures/fraggle-rock/reports/sample-flow-result.json [hunk 4]
```diff
@@ -3299,10 +3299,11 @@
           },
           "doctype": {
             "id": "doctype",
-            "title": "Page has the HTML doctype",
+            "title": "Page lacks the HTML doctype, thus triggering quirks-mode",
```
Comment (adamraine) [inline core/test/fixtures/fraggle-rock/reports/sample-flow-result.json:3302]: Looks like the doctype artifact is out of date on the flow artifacts:
`yarn update:flow-sample-json --rebaseline-artifacts Doctype`

Comment (connorjclark) [reply core/test/fixtures/fraggle-rock/reports/sample-flow-result.json:3302]: yeah thx

btw
image

File: core/test/fixtures/fraggle-rock/artifacts/sample-flow-artifacts.json [hunk 5]
```diff
@@ -295067,7 +295067,8 @@
         "Doctype": {
           "name": "html",
           "publicId": "",
-          "systemId": ""
+          "systemId": "",
+          "documentCompatMode": "CSS1Compat"
```
Comment (adamraine) [inline core/test/fixtures/fraggle-rock/artifacts/sample-flow-artifacts.json:295071]: This is only for one flow step, shouldn't there be more

Comment (connorjclark) [reply core/test/fixtures/fraggle-rock/artifacts/sample-flow-artifacts.json:295071]: I suppose! The update script didn't work so I just updated one thing 🙈

**Pull request merged**